src/Swarm/Game/Step.hs

```diff
@@ -1663,3 +1664,20 @@ updateAvailableRecipes invs e = do
       newCount = length newRecipes
   availableRecipes .= newRecipes <> knownRecipes
   availableRecipesNewCount += newCount
+  updateAvailableCommands e
+
+updateAvailableCommands :: Has (State GameState) sig m => Entity -> m ()
+updateAvailableCommands e = do
+  let newCaps = S.fromList (e ^. entityCapabilities)
+      keepConsts s
+        | S.null s = False
+        -- This presently only works for single capability command.
+        -- If a command needs more than one capability, then we need combine all the known capabilities
```

Comment:
I think we currently don't have any commands that require multiple capabilities, though in theory we could.

Reply:
Until then, I guess it would be safer to change `constCap` to provides a `Maybe Capability`

Reply:
Sure, that's a good idea.  I can't think of any compelling reason we really *need* commands that require multiple capabilities.  In fact, if we ever wanted a command that required capabilities A and B, we could just make a new capability C, and the device that provides capability C could be constructed out of the devices that provide capabilities A and B.  So I would be totally fine with enshrining that restriction in the types.